sql/moz-fx-data-shared-prod/fenix_derived/feature_usage_events_v1/query.sql

```diff
@@ -0,0 +1,1461 @@
+-- Query for fenix_derived.feature_usage_events_v1
+            -- For more information on writing queries see:
+            -- https://docs.telemetry.mozilla.org/cookbooks/bigquery/querying.html
+DECLARE start_date DATE DEFAULT "2021-01-01";
+
+DECLARE end_date DATE DEFAULT current_date;
+
+WITH dau_segments AS (
+  SELECT
+    DATE(submission_timestamp) AS submission_date,
+    COUNT(DISTINCT client_info.client_id) AS dau
+  FROM
+    mozdata.fenix.events_unnested
+      --AND channel = 'release'
+  WHERE
+    DATE(submission_timestamp) >= '2021-01-01'
+  GROUP BY
+    1
+),
+product_features AS (
+  SELECT
+    client_info.client_id,
+    DATE(submission_timestamp) AS submission_date,
+    /*Logins*/
+    --autofill
+    CASE
+      WHEN event_category = 'logins'
+        AND event_name = 'password_detected'
+        THEN 1
+      ELSE 0
+    END AS autofill_password_detected_logins,
+    CASE
+      WHEN event_category = 'logins'
+        AND event_name = 'autofill_prompt_shown'
+        THEN 1
+      ELSE 0
+    END AS autofill_prompt_shown_logins,
+    CASE
+      WHEN event_category = 'logins'
+        AND event_name = 'autofill_prompt_dismissed'
+        THEN 1
+      ELSE 0
+    END AS autofill_prompt_dismissed_logins,
+    CASE
+      WHEN event_category = 'logins'
+        AND event_name = 'autofilled'
+        THEN 1
+      ELSE 0
+    END AS autofilled_logins,
+    --management
+    CASE
+      WHEN event_category = 'logins'
+        AND event_name = 'management_add_tapped'
+        THEN 1
+      ELSE 0
+    END AS management_add_tapped_logins,
+    CASE
+      WHEN event_category = 'logins'
+        AND event_name = 'management_logins_tapped'
+        THEN 1
+      ELSE 0
+    END AS management_tapped_logins,
+    /*Credit Card*/
+    --autofill
+    CASE
+      WHEN event_category = 'credit_cards'
+        AND event_name = 'form_detected'
+        THEN 1
+      ELSE 0
+    END AS form_detected_cc,
+    CASE
+      WHEN event_category = 'credit_cards'
+        AND event_name = 'autofill_prompt_shown'
+        THEN 1
+      ELSE 0
+    END AS autofill_prompt_shown_cc,
+    CASE
```

Comment:
Could we make the query shorter and more readable if we used `COUNTIF(condition) AS field`?

Reply:
This applies to many of the fields in this query.